blackbox/docs/appendices/release-notes/4.0.0.rst

```diff
@@ -0,0 +1,424 @@
+.. _version_4.0.0:
+
+=============
+Version 4.0.0
+=============
+
+Released on 2019/06/25.
+
+.. NOTE::
+
+    If you are upgrading a cluster, you must be running CrateDB 3.0.4 or higher
+    before you upgrade to 4.0.0.
+
+    We recommend that you upgrade to the latest 3.3 release before moving to
+    4.0.0.
+
+    Any upgrade from a version prior to :ref:`version_4.0.0` will require
+    a `full restart upgrade`_.
+
+    When restarting, CrateDB will migrate indexes to a newer format. Depending
+    on the amount of data, this may delay node start-up time.
+
+    Please consult the :ref:`version_4.0.0_upgrade_notes` before upgrading.
+
+.. WARNING::
+
+    Tables that were created prior CrateDB 3.x will not function with 4.x
+    and must be recreated before moving to 4.x.x.
+
+    You can recreate tables using ``COPY TO`` and ``COPY FROM`` or by
+    `inserting the data into a new table`_.
+
+    Before upgrading, you should `back up your data`_.
+
+.. _full restart upgrade: http://crate.io/docs/crate/guide/best_practices/full_restart_upgrade.html
+.. _back up your data: https://crate.io/a/backing-up-and-restoring-crate/
+.. _inserting the data into a new table: https://crate.io/docs/crate/reference/en/latest/admin/system-information.html#tables-need-to-be-recreated
+
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+.. _version_4.0.0_upgrade_notes:
+
+Upgrade Notes
```

Comment:
what about the order of `Upgrade Notes` and `Changes`? In the 2.0.0 and 3.0.0 we put `Changes` first.

Reply:
I think `Upgrade Notes` are a bit more important, so I'd keep it like that.